opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -723,12 +723,12 @@ namespace Opm
         //  Set current control to bhp, and bhp value in state, set bhp limit in Well object.
         if (well_copy.well_ecl_.isInjector()) {
             auto prop = std::make_shared<Well::WellInjectionProperties>(well_copy.well_ecl_.getInjectionProperties());
-            prop->BHPLimit.reset(bhp);
+            prop->BHPLimit.reset(bhp/1e5); // HACK WARNING: internal unit is same as deck, assuming metric!
```

Comment:
Should have a compiler warning on this

Reply:
Addressed by refactoring as discussed offline: passing InjectionControls and ProductionControls as parameters to the equation-solving methods.